File: ios/BinancePay.m
#import <React/RCTBridgeModule.h>
#import <BinancePaySDK/BinancePaySDK.h>

OrderInitParameters * parameters;

@interface RCT_EXTERN_MODULE(BinancePay, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(initBinancePayParam:(NSString *)merchantId (NSString *)prepayId (NSString *)timeStamp (NSString *)nonceStr (NSString *)certSn (NSString *)sign)

@end

File: ios/BinancePay.swift
@objc(BinancePay)
class BinancePay: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
   @objc(initBinancePayParam:)
   func initBinancePayParam(merchantId: NSString, prepayId: NSString, timeStamp: NSString, nonceStr: NSString, certSn: NSString, sign: NSString) -> Void {
       self.parameters = OrderInitParameters(merchantId: merchantId, prepayId: prepayId, timestamp: timeStamp, noncestr: nonceStr, certSn: certSn, sign: sign, redirectScheme: nil)
   }
}
